deps/wazuh_testing/wazuh_testing/remote.py

```diff
@@ -479,6 +479,31 @@ def wait_to_remoted_key_update(wazuh_log_monitor):
     check_remoted_log_event(wazuh_log_monitor, callback_pattern, error_message, timeout=20)
 
 
+def wait_to_remoted_update_groups(wazuh_log_monitor):
+    """Allow to detect when remoted has reloaded its groups and multigroups.
+
+    This is necessary for remoted to find shared group files to send to agents.
+
+    The reload time is editable in the internal_options.conf and defaults to 10 seconds.
+
+    >> remoted.shared_reload=10
+
+    Args:
+        wazuh_log_monitor (FileMonitor): FileMonitor object to monitor the Wazuh log.
+
+    Raises:
+        TimeoutError: if could not find the remoted key loading log.
+    """
+    # We have to make sure that remoted has correctly reloaded its groups and multigroups.
+    # The log is truncated to ensure that the information has been loaded after the agent has been registered.
+    truncate_file(LOG_FILE_PATH)
+
+    callback_pattern = '.*c_files().*End updating shared files sums.'
+    error_message = 'Could not find the groups reload log'
+
+    check_remoted_log_event(wazuh_log_monitor, callback_pattern, error_message, timeout=20)
```

Comment:
It is recommended to remove hardcoded timeouts. Also, in other multigroups development https://github.com/wazuh/wazuh-qa/pull/2611/files#diff-ccad5ed309e3a3a6adf6c288967546dab4637eddedbba5155a8f2fc00bf0d930R28 the specified timeout is set to `10`. We need to ensure to use the same timeouts in order to have common parameters to check this functionality.

I recommend defining a common variable in the framework and use it in all multigroup testing